test_util/aws.py

```diff
@@ -0,0 +1,113 @@
+
+    def resource(self, name):
+        return self.session.resource(service_name=name, region_name=self.region)
+
+    def create_key_pair(self, key_name):
+        """Retruns private key of newly generated pair
+        """
+        key = self.resource('ec2').KeyPair(key_name)
+        return key.key_material
+
+    def delete_key_pair(self, key_name):
+        self.resource('ec2').KeyPair(key_name).delete()
+
+    def create_stack(self, name, template_url, user_parameters, deploy_timeout=60):
+        """Returns boto stack object
+        """
+        logging.info('Requesting AWS CloudFormation...')
+        cf_parameters = []
+        for k, v in user_parameters.items():
+            cf_parameters.append({'ParameterKey': k, 'ParameterValue': v})
+        self.resource('cloudformation').create_stack(
+            StackName=name,
+            TemplateURL=template_url,
+            DisableRollback=True,
+            TimeoutInMinutes=deploy_timeout,
+            Capabilities=['CAPABILITY_IAM'],
+            Parameters=cf_parameters)
+        return CFStack(name, self)
+
+
+class CFStack():
+    def __init__(self, stack_name, boto_wrapper):
+        self.boto_wrapper = boto_wrapper
+        self.stack = self.boto_wrapper.resource('cloudformation').Stack(stack_name)
+
+    def wait_for_stack_creation(self):
+        self.boto_wrapper.client('cloudformation').get_waiter('stack_create_complete').\
+            wait(StackName=self.stack.stack_name)
+
+    def wait_for_stack_deletion(self):
+        self.boto_wrapper.client('cloudformation').get_waiter('stack_delete_complete').\
+            wait(StackName=self.stack.stack_name)
+
+
+class DcosCfSimple(CFStack):
+    @classmethod
+    def create(cls, stack_name, template_url, public_agents, private_agents,
+               admin_location, key_pair_name, boto_wrapper):
+        parameters = {
+            'KeyName': key_pair_name,
+            'AdminLocation': admin_location,
+            'PublicSlaveInstanceCount': str(public_agents),
+            'SlaveInstanceCount': str(private_agents)}
+        boto_wrapper.create_stack(stack_name, template_url, parameters)
+        return cls(stack_name, boto_wrapper)
+
+    def delete(self):
+        logger.info('Starting deletion of CF stack')
+        # boto stacks become unusable after deletion (e.g. status/info checks) if name-based
+        self.stack = self.boto_wrapper.resource('cloudformation').Stack(self.stack.stack_id)
+        self.stack.delete()
+        bucket_id = self.stack.Resource('ExhibitorS3Bucket').physical_resource_id
+        s3 = self.boto_wrapper.resource('s3')
+        bucket = s3.Bucket(bucket_id)
+        logger.info('Starting bucket {} deletion'.format(bucket))
+        all_objects = bucket.objects.all()
+        obj_count = len(list(all_objects))
+        if obj_count > 0:
+            assert obj_count == 1, 'Expected one object in Exhibitor S3 bucket but found: ' + str(obj_count)
```

Comment:
random aside: cosmos is thinking of stashing things in this s3 bucket, we'll need to revisit if / when that happens.